.github/workflows/test_and_publish_charm.yaml

```diff
@@ -15,6 +15,40 @@ on:
       - track/**
 
 jobs:
-  publish-to-edge:
-    uses: canonical/operator-workflows/.github/workflows/test_and_publish_charm.yaml@main
+  lib-check:
+    # This job is the precondition of all jobs that need the Charmhub token.
+    # By disabling this job on forked repositories, we can achieve
+    # disabling the publishing to Charmhub action on forked repositories
+    # while still running tests on push events
+    if: github.repository_owner == 'canonical'
+    name: Check libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@2.1.1
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  unit-tests:
+    uses: canonical/operator-workflows/.github/workflows/test.yaml@main
+    secrets: inherit
+  integration-tests:
+    uses: ./.github/workflows/integration_test.yaml
+  select-channel:
+    name: Select target channel
+    runs-on: ubuntu-22.04
+    outputs:
+      channel: ${{ steps.channel.outputs.name }}
+    steps:
+      - name: Select charmhub channel
+        id: channel
+        uses: canonical/charming-actions/channel@2.1.1
+  publish-charm:
+    name: Publish
+    uses: canonical/operator-workflows/.github/workflows/publish_charm.yaml@main
+    needs: [unit-tests, integration-tests, select-channel]
+    with:
+      channel: ${{ needs.select-channel.outputs.channel }}
+      images: ${{ needs.integration-tests.outputs.images }}
```

Comment:
Not really sure I understand why changing this job.
If I get it right, it's because on fork repos, it'll try to run this and fail due to absence of charmhub token ?

If that's it, isn't it more relevant to make this change in operator-workflow repo directly ?

Reply:
The integration test suite for this requires argument flags such as `--num-units` and `--test-db-from-config` which need to run in a matrix, and currently there's no way pass these down to the reusable workflow.
Also, there are additional integration tests specific to this repository that requires specific secrets for testing `akismet` plugin and `openID` plugin.

So in the end, it runs the same sets of jobs as `test_and_publish_charm.yaml` in operator workflows, but refers to different set of integration tests that currently cannot be supported by operator workflows.

Would this information help the reason this workflow? Would be glad to discuss other alternatives!